components/src/dynamo/replay/world.py

```
@@ -0,0 +1,827 @@
+        self._closed = True
+        loop = self._loop
+        self._loop = None
+        if loop is None:
+            return
+
+        async def shutdown_sessions() -> list[BaseException]:
+            errors: list[BaseException] = []
+            for participant_id in sorted(self._sessions):
+                try:
+                    await self._sessions[participant_id].shutdown_async()
+                except BaseException as exc:  # Continue closing sibling engines.
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `BaseException`

(BLE001)